R/subscript-tables.R

```diff
@@ -389,5 +401,118 @@ nameDimensionAttributes <- function(x)
         names(dimnames.x) <- dim.names
         dimnames(x) <- dimnames.x
     }
+    # attr(x, "dim.types") <- type.per.dim
     return(x)
 }
+
+# Return the number of dimensions for a given question type.
+# This can either be 1 or 2.
+#' @param question.types A vector of characters corresponding to question types
+#' @noRd
+questionDimension <- function(question.types) {
+    q.dims <- rep(1, length(question.types))
+    q.dims[question.types %in% c("PickOneMulti", "PickAnyGrid", "NumberGrid")] <- 2
+    q.dims
+}
+
+# Determine new questiontypes by comparing the new QTable to the original
+#' @param new The newly-subscripted QTable
+#' @param original The original QTable pre-subscripting
+#' @noRd
+getUpdatedQuestionTypes <- function(new, original) {
```

Comment:
Not sure how this deals with multi-stat tables

Reply:
fixed with small change